src/FixedDataTableCellGroup.js

```diff
@@ -129,48 +210,71 @@ class FixedDataTableCellGroupImpl extends React.Component {
     /*object*/ cellTemplate,
     /*number*/ left,
     /*string*/ key,
-    /*number*/ columnGroupWidth,
-    /*boolean*/ isColumnReordering,
+    /*number*/ columnGroupWidth
   ) /*object*/ => {
 
-    var cellIsResizable = columnProps.isResizable &&
-      this.props.onColumnResize;
-    var onColumnResize = cellIsResizable ? this.props.onColumnResize : null;
-
-    var cellIsReorderable = columnProps.isReorderable && this.props.onColumnReorder && rowIndex === -1 && columnGroupWidth !== columnProps.width;
-    var onColumnReorder = cellIsReorderable ? this.props.onColumnReorder : null;
-
     var className = columnProps.cellClassName;
     var pureRendering = columnProps.pureRendering || false;
 
+    const onColumnReorderEndCallback =
+      (columnProps.isReorderable)
+        ? this.props.onColumnReorderEndCallback
+        : null;
+    const onColumnResizeEndCallback =
+    (columnProps.isResizable)
+      ? this.props.onColumnResizeEndCallback
+      : null;
+
     return (
       <FixedDataTableCell
         isScrolling={this.props.isScrolling}
         isHeaderOrFooter={this.props.isHeaderOrFooter}
+        isHeader={this.props.isHeader}
+        tableHeight={this.props.tableHeight}
         align={columnProps.align}
         className={className}
         height={height}
         key={key}
         maxWidth={columnProps.maxWidth}
         minWidth={columnProps.minWidth}
         touchEnabled={this.props.touchEnabled}
-        onColumnResize={onColumnResize}
-        onColumnReorder={onColumnReorder}
-        onColumnReorderMove={this.props.onColumnReorderMove}
-        onColumnReorderEnd={this.props.onColumnReorderEnd}
-        isColumnReordering={isColumnReordering}
-        columnReorderingData={this.props.columnReorderingData}
+        onColumnResizeEndCallback={onColumnResizeEndCallback}
+        onColumnReorderEndCallback={onColumnReorderEndCallback}
         rowIndex={rowIndex}
         columnKey={columnProps.columnKey}
         width={columnProps.width}
         left={left}
+        cellGroupLeft={this.props.left}
         cell={cellTemplate}
         columnGroupWidth={columnGroupWidth}
         pureRendering={pureRendering}
         isRTL={this.props.isRTL}
+        scrollX={this.props.scrollX}
+        isFixed={this.props.isFixed}
+        availableScrollWidth={this.props.availableScrollWidth}
+        maxScrollX={this.props.maxScrollX}
+        scrollToX={this.props.scrollToX}
+        toggleCellsRecycling={this.toggleCellsRecycling}
+        getCellGroupWidth={this.getCellGroupWidth}
       />
     );
   }
+
+  /**
+   * @deprecated
```

Comment:
put a comment on why this is marked for deprecation, and how we plan to overcome this?

Reply:
Done